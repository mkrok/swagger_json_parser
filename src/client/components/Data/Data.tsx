import * as React from 'react';
import styled from 'styled-components';
import { useAppData } from '~/client/context/AppStateProvider';
import Tag from '~/client/components/Tag';
import Info from '~/client/components/Info';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  justify-content: flex-start;
  background-color: #222;
  color: #ccc;
  padding: 0.5em;
`;

const Data = () => {
  const { appData } = useAppData();
  const { transformedTags } = appData;
  return (
    <Wrapper>
      {transformedTags && <Info info={appData.info} />}
      {transformedTags &&
        transformedTags.map((tag, index) => <Tag key={`tag=${index + 1}`} tag={{ ...tag }} />)}
    </Wrapper>
  );
};

export default Data;
