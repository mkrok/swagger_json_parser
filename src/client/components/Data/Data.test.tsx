import renderer from 'react-test-renderer';
import Data from './';

describe('Data', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<Data />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
