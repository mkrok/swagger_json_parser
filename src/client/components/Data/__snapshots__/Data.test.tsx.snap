// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data should match snapshot 1`] = `
<div
  className="sc-cyRfQX edZQld"
/>
`;
